From c2c190f127d51fe1067687aa7bb1cd26613ba914 Mon Sep 17 00:00:00 2001
From: Joerg Roedel <jroedel@suse.de>
Date: Thu, 21 Feb 2019 14:49:58 +0100
Subject: [PATCH] KVM: VMX: Missing part of upstream commit 904e14fb7cb9
Git-commit: 904e14fb7cb96401a7dc803ca2863fd5ba32ffe6
Patch-mainline: v4.16-rc1
References: bsc#1124166

Stable backport of upstream commit removed the check added
here because stable-4.4.y does not have support for per-vcpu
apicv disabling.

SLE12-SP3 adds support for this so we need to add the check
too.

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/kvm/vmx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 14553f6c03a6..e28b67e33300 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -4632,7 +4632,7 @@ static u8 vmx_msr_bitmap_mode(struct kvm_vcpu *vcpu)
 	    (vmcs_read32(SECONDARY_VM_EXEC_CONTROL) &
 	     SECONDARY_EXEC_VIRTUALIZE_X2APIC_MODE)) {
 		mode |= MSR_BITMAP_MODE_X2APIC;
-		if (enable_apicv)
+		if (enable_apicv && kvm_vcpu_apicv_active(vcpu))
 			mode |= MSR_BITMAP_MODE_X2APIC_APICV;
 	}
 
-- 
2.16.3

